torch/csrc/autograd/python_function.cpp

```diff
@@ -382,87 +356,101 @@ static void _transplant_var(VariableImpl& var, const std::shared_ptr<Function>&
 // do in this case.  After this method is run, t2var is extended with
 // mappings for output tensors as well.
 static void _wrap_outputs(THPFunction *self, t2var_type &t2var,
-    std::unordered_set<PyObject *> &dirty_inputs, PyObject *raw_output,
-    PyObject *outputs, bool is_volatile)
+    std::unordered_set<PyObject *> &dirty_inputs,
+    const t2var_type &shared_pairs,
+    PyObject *raw_output, PyObject *outputs, bool is_volatile)
 {
-  auto cdata = is_volatile ? nullptr : THPFunction_asFunction(self);
+  bool is_executable = self->cdata.is_executable && !is_volatile;
```

Comment:
Can you remove `!is_volatile`? The first condition should be enough (as discussed in the previous review)